### ISSUE TITLE
Add mask to input 

### ISSUE THREAD

**Issue by mheinzinger** [sacdallago/biotrainer#16]
When experimentally measuring the 3D structure of a protein sequence of length L, it can happen that not all L residues can be experimentally resolved. In such cases, we still know which amino acid was given in the input but we have no 3D coordinates for it. When we encode the 3D structure as a string of the 3 secondary elements "H" (helix), "B" ("beta-sheet), "O" (other), this means that we can not assign all L residues in the input sequence to one of those classes.
One easy solution would be to simply cut out the residues that were not resolved.
However, this should not be done as we a) create unrealisitic and potentially non-viable protein sequences when doing this  and b) our language models actually benefit from those additional residues in the input as they add context.
Therefor, I used a binary mask when training secondary structure. This mask has length L and has a 1 at positions that were resolved and a 0 for positions that were unresolved. During training, I compute the loss only over those elements that have a 1 and ignore those that have a 0. During evaluation, I compute all my metrics only over the subset of residues that have a 1 in the mask and ignore the ones with 0.
As PyTorch's loss functions already have a flag 'ignore_index' which allows you to easily ignore padded elements (by default this shuold be -100), you could also simplify the above scenario by simply mapping unresolved residues also to -100. In this case, I assume your current code should already handle those as it will neither user elements with -100 for loss-compution nor for computing evaluation metrics. 

Let's discuss this on Friday in more detail if you have any questions about this.
